dds_web/api/user.py

```diff
@@ -418,6 +418,55 @@ def post(self):
         )
 
 
+class RemoveUserAssociation(flask_restful.Resource):
+    @auth.login_required
+    def post(self):
+        """Remove a user from a project"""
+
+        args = flask.request.json
+
+        project_id = args.pop("project")
+        user_email = args.pop("email")
+
+        # Check if email is registered to a user
+        existing_user = user_schemas.UserSchema().load({"email": user_email})
+        project = project_schemas.ProjectRequiredSchema().load({"project": project_id})
+
+        if existing_user:
+            user_in_project = False
+            for user_association in project.researchusers:
+                if user_association.user_id == existing_user.username:
+                    user_in_project = True
+                    db.session.delete(user_association)
+            if user_in_project:
+                try:
+                    db.session.commit()
+                    message = (
+                        f"User with email {user_email} no longer associated with {project_id}."
+                    )
+                    status = 200
+                except (sqlalchemy.exc.SQLAlchemyError, sqlalchemy.exc.IntegrityError) as err:
+                    flask.current_app.logger.exception(err)
+                    db.session.rollback()
+                    message = "Removing user association with the project has not succeeded"
+                    raise ddserr.DatabaseError(message=f"Server Error: {message}")
+            else:
+                message = "User already not associated with this project"
+                status = 200
+
+            flask.current_app.logger.debug(
+                f"User {existing_user.username} no longer associated with project {project.public_id}."
+            )
+        else:
+            message = f"Error: {user_email} does not exist in the db."
+            status = ddserr.error_codes["NoSuchUserError"]["status"].value
```

Comment:
We shouldnt return that a user doesn't exist, instead return that "the user is not associated to this project now" (not exactly that text but just saying "sure, the user doesn't have access" )

Reply:
Changed it now.